multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/configuration/service/ObjectStoreServiceInfoCreator.java

```diff
@@ -97,25 +99,33 @@ private URL getContainerUriEndpoint(Map<String, Object> credentials) {
     }
 
     private ObjectStoreServiceInfo createServiceInfoForGcpCloud(Map<String, Object> credentials) {
-        String bucket = (String) credentials.get(Constants.BUCKET);
-        String region = (String) credentials.get(Constants.REGION);
-        Supplier<Credentials> credentialsSupplier = getGcpCredentialsSupplier(credentials);
+        String bucketName = (String) credentials.get(Constants.BUCKET);
+        Storage storage = createObjectStoreStorage(credentials);
         return ImmutableObjectStoreServiceInfo.builder()
                                               .provider(Constants.GOOGLE_CLOUD_STORAGE)
-                                              .credentialsSupplier(credentialsSupplier)
-                                              .container(bucket)
-                                              .region(region)
+                                              .container(bucketName)
+                                              .gcpStorage(storage)
                                               .build();
     }
 
-    protected Supplier<Credentials> getGcpCredentialsSupplier(Map<String, Object> credentials) {
+    public Storage createObjectStoreStorage(Map<String, Object> credentials) {
+        return StorageOptions.newBuilder()
+                             .setCredentials(getGcpCredentialsSupplier(credentials))
+                             .build()
+                             .getService();
+    }
```

Comment:
StorageOptions and GCP provider has a built in retry strategy.
We can utilize it and remove the custom retry strategies? I mean we can move Storage creation in the GCP client and in the info creator only to provide the necessary credentials. We're binding this class with gcp classes now.
What do you think?